external/fv3fit/fv3fit/pytorch/graph/train.py

```diff
@@ -82,97 +79,88 @@ def train_graph_model(
     hyperparameters: GraphHyperparameters,
     train_batches: tf.data.Dataset,
     validation_batches: Optional[tf.data.Dataset],
-):
+) -> PytorchModel:
     """
     Train a graph network.
 
     Args:
         hyperparameters: configuration for training
         train_batches: training data, as a dataset of Mapping[str, tf.Tensor]
+            where each tensor has dimensions [batch, time, tile, x, y(, z)]
         validation_batches: validation data, as a dataset of Mapping[str, tf.Tensor]
+            where each tensor has dimensions [batch, time, tile, x, y(, z)]
     """
-
-    # use transforms to get correct variables, correct dimensions
-    processed_train_batches = train_batches.map(apply_to_mapping(ensure_nd(3)))
-
+    train_batches = train_batches.map(apply_to_mapping(ensure_nd(6)))
     sample = next(
-        iter(
-            processed_train_batches.unbatch().batch(
-                hyperparameters.normalization_fit_samples
-            )
-        )
+        iter(train_batches.unbatch().batch(hyperparameters.normalization_fit_samples))
     )
 
-    normalizer = get_normalizer(sample)
+    scalers = get_scalers(sample)
+    mapping_scaler = get_mapping_scaler(scalers)
```

Comment:
It's hard because we can't really type hint on the sample type of tfdatasets. This is a scaler that works on tfdatasets with mapping samples as opposed to tensor samples. I'll see what I can do as far as renaming and adding comments/type hints to make it clearer.

Reply:
OK, I think I was able to make this clearer.